java/client/src/main/java/glide/api/models/commands/SetOptions.java

```diff
@@ -0,0 +1,174 @@
+            this.type = type;
+            this.count = count;
+        }
+
+        /**
+         * Retain the time to live associated with the key. Equivalent to <code>KEEPTTL</code> in the
+         * Redis API.
+         */
+        public static Expiry KeepExisting() {
+            return new Expiry(KEEP_EXISTING);
+        }
+
+        /**
+         * Set the specified expire time, in seconds. Equivalent to <code>EX</code> in the Redis API.
+         *
+         * @param seconds time to expire, in seconds
+         * @return Expiry
+         */
+        public static Expiry Seconds(Long seconds) {
+            return new Expiry(SECONDS, seconds);
+        }
+
+        /**
+         * Set the specified expire time, in milliseconds. Equivalent to <code>PX</code> in the Redis
+         * API.
+         *
+         * @param milliseconds time to expire, in milliseconds
+         * @return Expiry
+         */
+        public static Expiry Milliseconds(Long milliseconds) {
+            return new Expiry(MILLISECONDS, milliseconds);
+        }
+
+        /**
+         * Set the specified Unix time at which the key will expire, in seconds. Equivalent to <code>
+         * EXAT</code> in the Redis API.
+         *
+         * @param unixSeconds unix time to expire, in seconds
+         * @return Expiry
+         */
+        public static Expiry UnixSeconds(Long unixSeconds) {
+            return new Expiry(UNIX_SECONDS, unixSeconds);
+        }
+
+        /**
+         * Set the specified Unix time at which the key will expire, in milliseconds. Equivalent to
+         * <code>PXAT</code> in the Redis API.
+         *
+         * @param unixMilliseconds unix time to expire, in milliseconds
+         * @return Expiry
+         */
+        public static Expiry UnixMilliseconds(Long unixMilliseconds) {
+            return new Expiry(UNIX_MILLISECONDS, unixMilliseconds);
+        }
+    }
+
+    /** Types of value expiration configuration. */
+    @RequiredArgsConstructor
+    protected enum ExpiryType {
+        KEEP_EXISTING("KEEPTTL"),
+        SECONDS("EX"),
+        MILLISECONDS("PX"),
+        UNIX_SECONDS("EXAT"),
+        UNIX_MILLISECONDS("PXAT");
+
+        private final String redisApi;
+    }
+
+    /** String representation of {@link #returnOldValue} when set. */
+    public static final String RETURN_OLD_VALUE = "GET";
+
+    /**
+     * Converts SetOptions into a String[] to add to a {@link Command} arguments.
+     *
+     * @return String[]
+     */
+    public String[] toArgs() {
+        List<String> optionArgs = new ArrayList<>();
+        if (conditionalSet != null) {
+            optionArgs.add(conditionalSet.redisApi);
+        }
+
+        if (returnOldValue) {
+            optionArgs.add(RETURN_OLD_VALUE);
+        }
+
+        if (expiry != null) {
+            optionArgs.add(expiry.type.redisApi);
+            if (expiry.type != KEEP_EXISTING) {
+                if (expiry.count == null) {
```

Comment:
I think you can omit this check with new constructors

Reply:
I was thinking the same thing, but does it hurt to keep it..?

Reply:
Maybe turn into an assert?

Reply:
It is ok to keep it